crud/common/call.lua

```diff
@@ -12,7 +12,59 @@ local call = {}
 
 local DEFAULT_VSHARD_CALL_TIMEOUT = 2
 
-local function call_impl(vshard_call, func_name, func_args, opts)
+
+-- for testing
+call.internal = {}
+
+local function get_vshard_read_call_name(prefer_replica, balance)
+    dev_checks('?boolean', '?boolean')
+
+    if not prefer_replica and not balance then
+        return 'callro'
+    end
+
+    if not prefer_replica and balance then
+        return 'callbro'
+    end
+
+    if prefer_replica and not balance then
+        return 'callre'
+    end
+
+    -- prefer_replica and balance
+    return 'callbre'
+end
```

Comment:
Sorry, I don't like it.
`vshard.router.map_callrw` (https://github.com/tarantool/vshard/issues/147) is coming soon. And it will be a single way to guarantee consistency when we perform map-reduce over all storages.
It will lead us to introducing new option?

So, anyway we should stay a way to perform "rw" call for read operations.

Reply:
Added `mode` option that can be `read` or `write`. `write` mode performs `rw`, `read` performs `callr*` according to `prefer_replica` and `balance`.